### PR TITLE
Cleaning folders (add) after method fails implemented AEA-491

### PR DESCRIPTION
**Pull request by panasevychol**
## Proposed changes

Cleaning folders after method fails implemented
#930 

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/add.py [hunk 1]
```diff
@@ -148,18 +151,35 @@ def _add_item(click_context, item_type, item_public_id) -> None:
...
     if item_public_id in [DEFAULT_CONNECTION, DEFAULT_PROTOCOL, DEFAULT_SKILL]:
         source_path = _find_item_in_distribution(ctx, item_type, item_public_id)
         package_path = _copy_package_directory(
-            ctx, source_path, item_type, item_public_id.name, item_public_id.author
+            ctx,
+            source_path,
+            item_type,
+            item_public_id.name,
+            item_public_id.author,
+            dest_path,
```
Comment (DavidMinarsch) [inline aea/cli/add.py:167]: you are saving the `dest_path` on ctx, so probably don't need this arg

File: aea/cli/common.py [hunk 2]
```diff
@@ -470,8 +471,29 @@ def _try_get_item_target_path(
...
+
+
 def _copy_package_directory(
-    ctx: Context, package_path: Path, item_type: str, item_name: str, author_name: str
+    ctx: Context,
+    package_path: Path,
+    item_type: str,
+    item_name: str,
+    author_name: str,
+    dest: str,
```
Comment (DavidMinarsch) [inline aea/cli/common.py:496]: see above

File: aea/cli/common.py [hunk 3]
```diff
@@ -488,7 +510,6 @@ def _copy_package_directory(
     # copy the item package into the agent's supported packages.
     item_type_plural = item_type + "s"
     src = str(package_path.absolute())
-    dest = os.path.join(ctx.cwd, "vendor", author_name, item_type_plural, item_name)
     logger.debug("Copying {} modules. src={} dst={}".format(item_type, src, dest))
```
Comment (DavidMinarsch) [inline aea/cli/common.py:513]: Aren't you missing removal of dest from clean_paths here?

Comment (panasevychol) [reply aea/cli/common.py:513]: No. It's added to a clean_paths in `_add_item` and removed in decorator.

**Pull request merged**